src/components/visualizations/chart/highcharts/test/customConfiguration.spec.ts

```diff
@@ -621,6 +622,70 @@ describe('getCustomizedConfiguration', () => {
             const dataLabel = formatter.call(dataLabelPoint);
             expect(dataLabel).toBe(expectation);
         });
+
+        describe('percentage data label formatter', () => {
+
+            it('should return empty data labels with undefined percentage', () => {
+                const result = percentageDataLabelFormatter.call({});
+                expect(result).toBe('');
+            });
+
+            it('should format data labels to percentage for single axis chart implicitly', () => {
+                const dataLabel = {
+                    percentage: 55.55
+                };
+                const result = percentageDataLabelFormatter.call(dataLabel);
+                expect(result).toBe('55.55%');
+            });
+
+            it.each([
+                ['left', false],
+                ['right', true]
+            ])('should format data labels to percentage for %s single axis chart', (_: string, opposite: boolean) => {
+                const dataLabel = {
+                    percentage: 55.55,
+                    series: {
+                        chart: {
+                            yAxis: [{}]
+                        },
+                        yAxis: {
+                            opposite
+                        }
+                    }
+                };
+                const result = percentageDataLabelFormatter.call(dataLabel);
+                expect(result).toBe('55.55%');
+            });
+
+            it.each([
+                [
+                    'should format data labels to percentage for dual axis chart on primary axis',
```

Comment:
it is ok to have test case sentence completely defined here, but you can also use something like 
```            it.each([
                ['', 'primary', false, '55.55%'],
                [' not', 'secondary', true, '123']
            ])('should%s format data labels to percentage for dual axis chart on %s axis', (_negation: string, _axis: string, opposite: boolean, expected: string) => {
```

Reply:
my first point is to avoid putting more params to `function`.
but we can try your new approach :)